ksqldb-common/src/test/java/io/confluent/ksql/metrics/StreamsErrorCollectorTest.java

```diff
@@ -56,73 +52,108 @@ public void setUp() {
         greaterThan(0.0));
   }
 
   @Test
   public void shouldComputeTopicLevelErrorStats() {
+    final MetricCollectors metricCollectors = new MetricCollectors();
+    final StreamsErrorCollector collector = new StreamsErrorCollector(
+        applicationId,
+        metricCollectors
+    );
+
     // Given:
     final String otherTopic = "other-topic";
     final int nmsgs = 3;
 
     // When:
-    IntStream.range(0, nmsgs).forEach(i -> recordError(applicationId, TOPIC_NAME));
-    IntStream.range(0, nmsgs + 1).forEach(i -> recordError(applicationId, otherTopic));
+    IntStream.range(0, nmsgs).forEach(i -> collector.recordError(TOPIC_NAME));
+    IntStream.range(0, nmsgs + 1).forEach(i -> collector.recordError(otherTopic));
 
     // Then:
-    final Collection<Stat> stats = MetricCollectors.getStatsFor(TOPIC_NAME, true);
+    final Collection<Stat> stats = metricCollectors.getStatsFor(TOPIC_NAME, true);
     final Optional<Stat> stat = stats.stream().filter((c) -> c.name().equals(CONSUMER_FAILED_MESSAGES)).findFirst();
     assertThat(stat.get().getValue(), equalTo(nmsgs * 1.0));
-    final Collection<Stat> otherStats = MetricCollectors.getStatsFor(otherTopic, true);
+    final Collection<Stat> otherStats = metricCollectors.getStatsFor(otherTopic, true);
     final Optional<Stat> otherStat = otherStats.stream().filter((c) -> c.name().equals(CONSUMER_FAILED_MESSAGES)).findFirst();
     assertThat(otherStat.get().getValue(), equalTo((nmsgs + 1) * 1.0));
   }
 
   @Test
   public void shouldComputeIndependentErrorStatsForQuery() {
-    // Given:
+    final MetricCollectors metricCollectors = new MetricCollectors();
+    final StreamsErrorCollector collector = new StreamsErrorCollector(
+        applicationId,
+        metricCollectors
+    );
+
     final String otherAppId = buildApplicationId();
+    final StreamsErrorCollector otherCollector = new StreamsErrorCollector(
+        otherAppId,
+        metricCollectors
+    );
+
     final String otherTopicId = "other-topic-id";
     final int nmsgs = 3;
-    IntStream.range(0, nmsgs).forEach(i -> recordError(applicationId, TOPIC_NAME));
-    IntStream.range(0, nmsgs + 1).forEach(i -> recordError(otherAppId, otherTopicId));
+    IntStream.range(0, nmsgs).forEach(i -> collector.recordError(TOPIC_NAME));
+    IntStream.range(0, nmsgs + 1).forEach(i -> otherCollector.recordError(otherTopicId));
 
     // When:
-    notifyApplicationClose(otherAppId);
+    otherCollector.cleanup();
 
     // Then:
+    final Collection<Stat> statsFor = metricCollectors.getStatsFor(TOPIC_NAME, true);
+    assertThat(statsFor, hasSize(2));
+    assertThat(getStat(statsFor, "consumer-failed-messages"), of(equalTo(nmsgs * 1.0)));
+    assertThat(getStat(statsFor, "consumer-failed-messages-per-sec"), of(greaterThan(0.0)));
+
+    final Collection<Stat> otherStatsFor = metricCollectors.getStatsFor(otherTopicId, true);
+    assertThat(otherStatsFor, hasSize(0));
+
```

Comment:
I kept a few checks I had added while I was trying to figure out what the test was trying to do.